lib/simp/cli/config/items/action/update_os_yum_repositories_action.rb

```diff
@@ -61,7 +61,11 @@ def apply
           # disable any CentOS repo spam
           if ! Dir.glob('CentOS*.repo').empty?
             debug( "Disabling CentOS repositories in #{@yum_repos_d}" )
-            execute( %q{grep "\\[*\\]" *CentOS*.repo | cut -d "[" -f2 | cut -d "]" -f1 | xargs yum-config-manager --disable} )
+            # Don't use pipes with spawn
+            repos = run_command(%q{grep "\\[*\\]" *CentOS*.repo})[:stdout].strip.split("\n")
```

Comment:
On my server, when the grep is applied to /etc/yum.repos.d/CentOS-Media.repo, the following lines match:
/etc/yum.repos.d/CentOS-Media.repo:#  yum --enablerepo=c7-media [command]
/etc/yum.repos.d/CentOS-Media.repo:#  yum --disablerepo=\* --enablerepo=c7-media [command]
/etc/yum.repos.d/CentOS-Media.repo:[c7-media]

The first two lines will result in garbage to be passed to the yum-config-manager command.  May want to use a regex that searches for [ at the beginning of the line to prevent this problem.